apis/v1beta1/types.go

```diff
@@ -14,12 +14,41 @@ import (
 type ProviderConfigSpec struct {
 	// Credentials required to authenticate to this provider.
 	Credentials ProviderCredentials `json:"credentials"`
+
+	// ClientID is the user-assigned managed identity's ID
+	// when Credentials.Source is `InjectedIdentity`. If unset and
+	// Credentials.Source is `InjectedIdentity`, then a system-assigned
+	// managed identity is used.
+	// Required if Credentials.Source is OIDCTokenFile.
+	// +kubebuilder:validation:Optional
+	ClientID *string `json:"clientID,omitempty"`
+
+	// TenantID is the Azure AD tenant ID to be used.
+	// If unset, tenant ID from Credentials will be used.
+	// Required if Credentials.Source is InjectedIdentity, OIDCTokenFile.
+	// +kubebuilder:validation:Optional
+	TenantID *string `json:"tenantID,omitempty"`
+
+	// MSIEndpoint is the optional path to a custom endpoint for
+	// Managed Service Identity.
+	// +kubebuilder:validation:Optional
+	MSIEndpoint *string `json:"msiEndpoint,omitempty"`
+
+	// The Cloud Environment which should be used. Possible values are "public",
+	// "usgovernment", "german", and "china". Defaults to "public".
+	// +kubebuilder:validation:Optional
+	Environment *string `json:"environment,omitempty"`
+
+	// OIDCTokenFilePath is the optional path to a token file
+	// that allows to access a managed identity.
+	// +kubebuilder:validation:Optional
+	OidcTokenFilePath *string `json:"oidcTokenFilePath,omitempty"`
 }
 
 // ProviderCredentials required to authenticate.
 type ProviderCredentials struct {
 	// Source of the provider credentials.
-	// +kubebuilder:validation:Enum=None;Secret;InjectedIdentity;Environment;Filesystem
+	// +kubebuilder:validation:Enum=None;Secret;UserAssignedManagedIdentity;SystemAssignedManagedIdentity;OIDCTokenFile
```

Comment:
I think it is better to keep the old methods ​​for compatibility.

```suggestion
	// +kubebuilder:validation:Enum=None;Secret;UserAssignedManagedIdentity;SystemAssignedManagedIdentity;OIDCTokenFile;InjectedIdentity;Environment;Filesystem
```